### ISSUE TITLE
Can't hide field of array element

### ISSUE THREAD

**Issue by Santinell** [mblarsen/mongoose-hidden#82]
Hello, I added `hide: true` to field of array element and find out that plugin don't hide such fields.
For example in object `{ name: 'Default', hosts: [{ name: 'Main', login: 'root', password: '123456' }]  }`
I can't hide password or any other field inside array.

```
    "mongoose": "^5.9.9"
    "mongoose-hidden": "^1.8.1"
```

How to reproduce:

```
const mongoose = require('mongoose');
const mongooseHidden = require('mongoose-hidden')();

async function bootstrap () {
  await mongoose.connect(process.env.MONGO_URL, {
    poolSize: 5,
    promiseLibrary: Promise,
    useNewUrlParser: true,
    useUnifiedTopology: true,
    useFindAndModify: false,
    useCreateIndex: true,
    keepAliveInitialDelay: 300000,
    connectTimeoutMS: 30000
  });

  const host = new mongoose.Schema({
    name: {
      type: String,
      hide: true
    },
    login: {
      type: String,
      hide: true
    },
    password: {
      type: String,
      hide: true
    }
  });

  const ProjectSchema = new mongoose.Schema({
    name: String,
    hosts: [host]
  }, {
    timestamps: { createdAt: 'created', updatedAt: 'modified' }
  });
  ProjectSchema.set('toObject', { virtuals: true });
  ProjectSchema.set('toJSON', { virtuals: true });
  ProjectSchema.plugin(mongooseHidden);
  const Project = mongoose.model('Project', ProjectSchema);

  await Project.create({
    name: 'Default',
    hosts: [
      { name: 'Main', login: 'root', password: '123456' },
      { name: 'Secondary', login: 'user', password: '654321' }
    ]
  });

  const project = await Project.findOne({ name: 'Default' });
  for (const host of project.hosts) {
    const obj = host.toObject();
    console.log(obj);
    if (obj.password) {
      throw new Error('host.password should not be defined');
    }
  }
}

bootstrap();
```

**Comment by mblarsen** [comment]
Hi @Santinell. Thanks for reporting the issue. It would be super helpful if you do a PR with a test case that encapsulates the issue.

You can add it to https://github.com/mblarsen/mongoose-hidden/blob/master/test/github-issues.js you can take inspiration from that and the other tests in the folder.

**Comment by mblarsen** [comment]
I've published version 1.9.0 that that fixes this (see #83)

**Comment by Santinell** [comment]
Thanks! It works!
